client/src/components/documents/Preview.tsx

```diff
@@ -0,0 +1,53 @@
+import { Dialog, DialogPanel, Button } from "@tremor/react";
+import { Viewer, Worker, ViewMode } from '@react-pdf-viewer/core';
+import { RiFileCloseFill } from "@remixicon/react";
+import { KxDocument } from "../../model";
+import { useEffect, useState } from "react";
+import API from "../../API";
+import { toast } from "../../utils/toaster";
+import { mongoose } from "@typegoose/typegoose";
+import '@react-pdf-viewer/core/lib/styles/index.css';
+
+export default function PreviewDoc(open: boolean, setOpen: (bool: boolean) => void, doc: KxDocument) {
+
+    const [fileUrl, setFileUrl] = useState<string | undefined>(undefined);
+
+    useEffect(() => {
+        const fetchDocuments = async () => {
+            try {
+                if (doc && doc._id) {
+                    const url = await API.getKxFileByID(new mongoose.Types.ObjectId("6730daada436c299749f589e"), "doc_63_info.pdf");
+                    setFileUrl(url.presignedUrl);
+                }
+            } catch (error) {
+                console.error(error);
+                toast({
+                    title: "Error",
+                    description: "Failed to retrieve file",
+                    variant: "error",
+                    duration: 3000,
+                })
+            }
+        };
+        fetchDocuments();
+    }, [doc]);
+
+    return (
+        <Dialog open={open} onClose={() => setOpen(false)} static={true}>
+            <DialogPanel>
+                <div className="flex justify-start items-center mb-4">
+                    <Button onClick={() => setOpen(false)} icon={RiFileCloseFill} className="mr-6"></Button>
```

Comment:
I think the position of the close button is not right. also putting a title can guide the user better. Generally, opening the modal just for showing one page and showing the link, is not necessary. is it possible to show the whole pages using navigation icon or remove the modal and direct the user to a new tab for showing the whole?

Reply:
that document is just one page, otherwise it shows the whole doc